### PR TITLE
MINIFICPP-1330-add conversion from microseconds

### PR DESCRIPTION
**Pull request by aminadinari19**
Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [Y] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [Y] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [Y] Has your PR been rebased against the latest commit within the target branch (typically main)?

- [Y] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI results for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Review by szaszm — APPROVED** [review]
Looks good to me

- [x] Tip: use 'x' inside the brackets to create a checked checkbox

File: libminifi/include/utils/ValueParser.h [hunk 1]
```diff
@@ -216,6 +216,7 @@ bool StringToTime(const std::string& input, Out& output, core::TimeUnit& timeuni
   }
 }
 
+
```
Comment (lordgamez) [inline libminifi/include/utils/ValueParser.h:219]: Minor: unnecessary ws.

File: libminifi/test/unit/PropertyTests.cpp [hunk 2]
```diff
@@ -22,6 +22,39 @@
...
+      return ParsingStatus::ParsingSuccessful;
+    }
+  } else {
+    return ParsingStatus::ParsingFail;
+  }
+}
+
+bool conversionTest(uint64_t number, core::TimeUnit unit, uint64_t check, ConversionTestTarget conversionUnit) {
+  uint64_t out;
+  bool returnStatus;
```
Comment (lordgamez) [inline libminifi/test/unit/PropertyTests.cpp:46]: Minor: I know these will be eventually filled by the parser function, but I would recommend giving an initial value for the primitive types.

File: libminifi/test/unit/PropertyTests.cpp [hunk 3]
```diff
@@ -22,6 +22,39 @@
...
+  if (conversionUnit == ConversionTestTarget::NS) {
+    returnStatus = org::apache::nifi::minifi::core::Property::ConvertTimeUnitToNS(number, unit, out);
+  } else if (conversionUnit == ConversionTestTarget::MS) {
+    returnStatus = org::apache::nifi::minifi::core::Property::ConvertTimeUnitToMS(number, unit, out);
+  }
+  if (returnStatus && out == check) {
+    return true;
+  } else {
+    return false;
+  }
```
Comment (lordgamez) [inline libminifi/test/unit/PropertyTests.cpp:56]:
```suggestion
 return returnreturnStatus && out == check;
```

File: libminifi/test/unit/PropertyTests.cpp [hunk 4]
```diff
@@ -22,6 +22,39 @@
 #include "../../include/core/Property.h"
 #include "utils/StringUtils.h"
 #include "../TestBase.h"
+namespace {
```
Comment (lordgamez) [inline libminifi/test/unit/PropertyTests.cpp:25]: Do we need the anonymous namespace here?

File: libminifi/include/core/Property.h [hunk 5]
```diff
@@ -180,10 +180,15 @@ class Property {
 // Compare
   bool operator <(const Property & right) const;
 
-// Convert TimeUnit to MilliSecond
```
Comment (lordgamez) [inline libminifi/include/core/Property.h:183]: :+1: I'm always for the idea of removing trivial comments

**Review by lordgamez — COMMENTED** [review]
LGTM, only a few minor comments

**Comment by aminadinari19** [comment]
Thank you. Will implement that tip in the future :)

File: libminifi/test/unit/PropertyTests.cpp (hunk 4 above)
Comment (hunyadi-dev) [reply libminifi/test/unit/PropertyTests.cpp:25]: It is a good practice to put all tests into anon namespaces.

File: libminifi/test/unit/PropertyTests.cpp (hunk 2 above)
Comment (aminadinari19) [reply libminifi/test/unit/PropertyTests.cpp:46]: Yeah that is always a better method. Will keep that in mind :)

File: libminifi/test/unit/PropertyTests.cpp (hunk 4 above)
Comment (lordgamez) [reply libminifi/test/unit/PropertyTests.cpp:25]: Thanks, I was just wondering if it was due to a linker error or some other reason.

**Review by lordgamez — APPROVED** [review]

**Pull request closed without merging**